web/src/pages/Leaderboard.js

```diff
@@ -9,12 +9,19 @@ export default function Leaderboard() {
   const navigate = useNavigate();
 
   const convertData = (obj, type, deltaObj) => {
-    return Object.entries(obj).map(([key, value], index) => ({
-      id: (index + 1).toString(),
-      name: key.replace(`${type}_`, ""),
-      score: value.toFixed(4),
-      improved: deltaObj[key] > 0.0 ? 1 : deltaObj[key] < 0.0 ? -1 : 0,
-    }));
+    const emailDomains = ["@gmail", "@hotmail", "@outlook", "@yahoo", "@icloud"];
```

Comment:
I think we should try to match the email pattern with some regular expression.

Reply:
just made some changes in the latest commit !